x-pack/plugins/enterprise_search/public/applications/shared/telemetry/send_telemetry.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useContext, useEffect } from 'react';
+
+import { HttpHandler } from 'src/core/public';
+import { KibanaContext, IKibanaContext } from '../../index';
+
+interface ISendTelemetryProps {
+  action: 'viewed' | 'error' | 'clicked';
+  value: string; // e.g., 'setup_guide'
+}
+
+interface ISendTelemetry extends ISendTelemetryProps {
+  http(): HttpHandler;
+  product: 'app_search' | 'workplace_search' | 'enterprise_search';
+}
+
+/**
+ * Base function - useful for non-component actions, e.g. clicks
+ */
+
+export const sendTelemetry = ({ http, product, action, metric }: ISendTelemetry) => {
```

Comment:
Looking at the definitions above... `ISendTelemetry` should have attributes `action`, `value`, `http`, and `product`, but I see us referencing `metric` here and not `value`, as defined. Am I missing something?

Reply:
Nope, this was me missing a var name change 🤦‍♀️ Rich suggested changing my generic `value` key to `metric`, and I totally missed this one (also, see my earlier question about me feeling like Typescript isn't actually catching anything? haha)

Reply:
Fixed: https://github.com/constancecchen/kibana/pull/5/files#diff-5bf73fdb46d840f67b85019bef1e0262R14

Reply:
Haha, yeah, I thought the same thing when I saw this ... like shouldn't TS have caught this?